Sources/TuistKit/Services/RunService.swift

```diff
@@ -0,0 +1,142 @@
+            return "Could't find simulator"
+        case let .targetNotFound(forSchemeName):
+            return "Couldn't find \(forSchemeName)"
+        }
+    }
+
+    var type: ErrorType {
+        switch self {
+        case .appNotFound, .simulatorNotFound, .targetNotFound:
+            return .abort
+        }
+    }
+}
+
+final class RunService {
+    let simulatorController: SimulatorControlling
+
+    let buildGraphInspector: BuildGraphInspecting
+
+    let projectGenerator: ProjectGenerating
+
+    let xcodebuildController: XcodeBuildControlling
+
+    var appPath: AbsolutePath?
+
+    var buildableTarget: Target?
+
+    init(simulatorController: SimulatorControlling = SimulatorController(),
+         buildGraphInspector: BuildGraphInspecting = BuildGraphInspector(),
+         projectGenerator: ProjectGenerating = ProjectGenerator(),
+         xcodebuildController: XcodeBuildControlling = XcodeBuildController())
+    {
+        self.simulatorController = simulatorController
+        self.buildGraphInspector = buildGraphInspector
+        self.projectGenerator = projectGenerator
+        self.xcodebuildController = xcodebuildController
+    }
+
+    func run(schemeName: String) throws {
+        try buildScheme(schemeName)
+
+        guard let buildableTarget = buildableTarget else {
+            throw RunServiceError.targetNotFound(forSchemeName: schemeName)
+        }
+
+        appPath = try findApp(for: buildableTarget)
+        try launchApp(for: buildableTarget)
+    }
+
+    private func buildScheme(_ schemeName: String) throws {
+        let graph = try projectGenerator.load(path: FileHandler.shared.currentPath)
+        guard let scheme = buildGraphInspector.buildableSchemes(graph: graph).first(where: { $0.name == schemeName }) else {
+            throw RunServiceError.appNotFound(targetName: "")
+        }
+
+        buildableTarget = buildGraphInspector.buildableTarget(scheme: scheme, graph: graph)
+
+        let workspacePath = try buildGraphInspector.workspacePath(directory: FileHandler.shared.currentPath)!
+
+        _ = try xcodebuildController.build(.workspace(workspacePath),
+                                           scheme: schemeName,
+                                           clean: false,
+                                           arguments: buildGraphInspector.buildArguments(target: buildableTarget!,
+                                                                                         configuration: nil))
+            .printFormattedOutput()
+            .toBlocking()
+            .last()
+    }
+
+    private func findApp(for target: Target) throws -> AbsolutePath {
+        let appsLocation = FileHandler.shared.locateDirectory("../Library/Developer/Xcode/DerivedData/",
+                                                              traversingFrom: FileHandler.shared.currentPath)?
+            .glob("\(target.name)-*")
+            .first?
+            .glob("Build/Products/Debug-iphonesimulator/*.app")
```

Comment:
With `Debug-iphonesimulator` we are assuming that the target is for iOS, which might not be true. It's worth getting the right path based on the target's platform. I think you can use [this method](https://github.com/tuist/tuist/blob/master/Sources/TuistCore/Models/Platform.swift#L75) to get the information that you need. 

Also, I'd introduce a check that exits the execution if the target is for a platform other than iOS. We can tell the user that we only support iOS at the moment.